packages/flutter_tools/lib/src/android/application_package.dart

```diff
@@ -372,6 +429,48 @@ class ApkManifestData {
       }
     }
 
+    // Handle case when use activity-alias
+    if (launchActivity == null) {
+      final Iterable<_Element> activityAliases = application.allElements('activity-alias');
+
+      for (final _Element activityAlias in activityAliases) {
+        final _Attribute? enabled = activityAlias.firstAttribute('android:enabled');
+        final Iterable<_Element> intentFilters = activityAlias.allElements('intent-filter');
+        final bool isEnabledByDefault = enabled == null;
+        final bool isExplicitlyEnabled = enabled != null && (enabled.value?.contains('0xffffffff') ?? false);
+        if (!(isEnabledByDefault || isExplicitlyEnabled)) {
+          continue;
+        }
+
+        for (final _Element element in intentFilters) {
+          final bool isMainAction = _isAttributeWithValuePresent(
+              element, 'action', 'android:name', '"android.intent.action.MAIN"');
+          if (!isMainAction) {
+            continue;
+          }
+          final bool isLauncherCategory = _isAttributeWithValuePresent(
+              element, 'category', 'android:name',
+              '"android.intent.category.LAUNCHER"');
+          if (!isLauncherCategory) {
+            continue;
+          }
+          launchActivity = activityAlias;
+          isActivityAlias = true;
+          break;
+        }
+        if (launchActivity != null) {
+          break;
+        }
+      }
+    }
+
+    // For checking if Target Activity is declared
+    Iterable<String?> activityNames = <String?>[];
+
+    if (isActivityAlias) {
+      activityNames = activities.map(_getActivityName);
+    }
```

Comment:
We can move this logic closer to where the results are used. Consider this set of changes: https://github.com/andrewkolos/flutter/commit/f46b251a54cab8ce6625abf226978376551514f6